javascript/ql/src/Statements/UselessConditional.ql

```diff
@@ -120,9 +120,12 @@ predicate whitelist(Expr e) {
  */
 predicate isConditional(ASTNode cond, Expr e) {
   e = cond.(IfStmt).getCondition() or
+  e = cond.(WhileStmt).getExpr() or
+  e = cond.(ForStmt).getTest() or
   e = cond.(ConditionalExpr).getCondition() or
-  e = cond.(LogAndExpr).getLeftOperand() or
-  e = cond.(LogOrExpr).getLeftOperand()
+  e = cond.(LogicalBinaryExpr).getLeftOperand() or
+  // Include `z` in `if (x && z)`.
+  isConditional(_, cond) and e = cond.(LogicalBinaryExpr).getRightOperand()
```

Comment:
I think we should squeeze this syntactic pattern a bit more: `cond.getUnderlyingValue().(LogicalBinaryExpr).getRightOperand()`.